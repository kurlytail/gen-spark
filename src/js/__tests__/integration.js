import { execSync } from 'child_process';

describe('# integration test', () => {
    beforeEach(() => {
        execSync('rm -rf testoutput');
    });

    it('## should print help options', () => {
        let output = execSync('npm run build').toString();
        output = execSync('sgen -g `pwd`/dist/spark.min.js -h').toString();
        expect(output).toMatchSnapshot();
    });

    it('## should generate design and run spark commands', () => {
        let output = execSync('npm run build').toString();
        output = execSync('sgen -g `pwd`/dist/spark.min.js -d src/test/fixture/design.json -o testoutput').toString();
        output = output.replace(/info: Loaded generator .*spark.min.js.*/, '');
        expect(output).toMatchSnapshot();
        execSync('npm install', { cwd: 'testoutput', stdio: 'inherit' });
        execSync('npm run lint', { cwd: 'testoutput', stdio: 'inherit' });
        execSync('mvn compile', { cwd: 'testoutput', stdio: 'inherit' });
    });
});
