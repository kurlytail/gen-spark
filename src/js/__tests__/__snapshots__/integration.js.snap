// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# integration test ## should generate design 1`] = `
"info: loading map templates/map.json
info: Initializing git repository testoutput/.git
info: Generated testoutput/.prettierrc from template templates/.prettierrc._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should generate design and run spark commands 1`] = `
"info: loading map templates/map.json
info: Initializing git repository testoutput/.git
info: Generated testoutput/.prettierrc from template templates/.prettierrc._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should generate design with merge 1`] = `
"info: loading map templates/map.json
info: Initializing git repository testoutput/.git
info: Generated testoutput/.prettierrc from template templates/.prettierrc._
info: Committed generated files
info: Attempting to use branch master-sgen-generated
info: Creating new branch master-sgen-generated
warn: Checkout branch master
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should generate design with merge 2`] = `
"info: loading map templates/map.json
info: Opening git repository testoutput/.git
warn: Ignoring error Error: cannot stash changes - there is nothing to stash.
info: Attempting to use branch master-sgen-generated
warn: Checkout branch master-sgen-generated
info: Generated testoutput/.prettierrc from template templates/.prettierrc._
warn: Checkout branch master-sgen-generated
warn: Checkout branch master-sgen-generated
warn: Nothing changed in generated files
warn: Checkout branch master
info: sgen is done
"
`;

exports[`# integration test ## should print help options 1`] = `
"Usage:
  node sgen [OPTION]

Options:
  -m, --map=ARG+       mapping files
  -d, --design=ARG+    design files
  -e, --extension=ARG+ extension directories or files
  -o, --output=ARG     output directory for generated files
      --overwrite=ARG  Overwrite modes, can be one of skip, regen, error                                  (default: error)
      --forceOverwrite Override overwrite flags in the generator map with the one provided on --overwrite
  -h, --help           display this help
"
`;
